server/handler/issue_comment.go

```diff
@@ -100,6 +100,10 @@ func (h *IssueComment) Handle(ctx context.Context, eventType, deliveryID string,
 		return nil
 	}
 
+	if !h.affectsApproval(event.GetComment().GetBody(), fc.Config.ApprovalRules) {
```

Comment:
I think this misses an edge case with edited comments. The previous `detectAndLogTampering` call exits early if the edit was by the same user that authored the original comment. Since `affectsApproval` is only looking at the current body, if a user edits their own comment to remove an approval, I think this will cause us to skip evaluating the PR (only the old body affects approval, and we don't check that.)

To catch that case, I think we need to look at the previous content for `edited` events as well, possibly assuming that a comment affects approval if the old content is missing (e.g. in #379, where a GHE bug removed this field.)

As a general rule, I want to make sure this type of optimization will over-evaluate if there are errors or uncertainty, rather than under-evaluate.